### ISSUE TITLE
consul provider requires values for ca_file, cert_file, and key_file

### ISSUE THREAD

**Issue by dansteen** [hashicorp/terraform#8499]
Hey!

Starting in 0.7.2 consul has started requiring values for ca_file, cert_file, and key_file.   If they are not filled in in the provider, then you are prompted for them on the command line.   This is the case even if you have specified a schema of "http".

For example, this config:
```
provider "consul" {
  address    = "${var.consul_address}:8500"
  datacenter = "awse"
  scheme     = "http"
  token = "${var.consul_token}"
}
```

Will prompt for the following values:

```
[~/projects/terraform-aws]$ terraform plan
provider.consul.ca_file
  Enter a value:  blah

provider.consul.cert_file
  Enter a value: blahblah

provider.consul.key_file
  Enter a value: blahblahblah
```

This config will run without prompting:

```
provider "consul" {
  address    = "consul.prod.awse.traitify.com:8500"
  datacenter = "awse"
  scheme     = "http"
  token = "${var.consul_token}"
  ca_file = "blah"
  cert_file = "blahblah"
  key_file = "blahblahblah"
}
```

Note, that there are no actual files called blah*

**Comment by daveadams** [comment]
I noticed this starting with 0.7.1. It also prompts for provider.consul.token (which I specify via the CONSUL_HTTP_TOKEN env var). But regardless, the docs say those fields are optional, and in fact if I just hit enter on each prompt everything works fine.

**Comment by daveadams** [comment]
PR submitted in #8551. The provider config was setting missing values to nil, which triggered the provider parser to prompt for values.

**Comment by lhdomenech** [comment]
0.7.2 also prompts for scheme and datacenter even though they are documented as optional.

**Comment by stack72** [comment]
Closed via #8551
